corehq/apps/events/forms.py

```diff
@@ -235,3 +235,63 @@ def __init__(self, *args, **kwargs):
                 data_bind="value: name, valueUpdate: 'keyup'",
             )
         )
+
+
+class EditAttendeeForm(forms.ModelForm):
+
+    class Meta:
+        model = AttendeeModel
+        fields = ('name', 'locations', 'primary_location')
+        help_texts = {
+            'locations': ExpandedMobileWorkerFilter.location_search_help,
+        }
+        widgets = {
+            'primary_location': PrimaryLocationWidget(
+                css_id='id_primary_location',
+                source_css_id='id_locations',
+            ),
+        }
+
+    def __init__(self, *args, domain, **kwargs):
+        self.domain = domain
+        self.base_fields['locations'].widget = LocationSelectWidget(
+            self.domain, multiselect=True, id='id_locations',
+        )
+
+        self.helper = FormHelper()
+        self.helper.form_method = 'POST'
+        self.helper.form_class = 'form-horizontal'
+        self.helper.form_tag = False
+        self.helper.label_class = 'col-sm-3 col-md-2'
+        self.helper.field_class = 'col-sm-9 col-md-8 col-lg-6'
+
+        super().__init__(*args, **kwargs)
+
+    def clean_locations(self):
+        location_ids = self.data.getlist('locations')
+        try:
+            locations = get_locations_from_ids(location_ids, self.domain)
+        except SQLLocation.DoesNotExist:
+            raise ValidationError(
+                _('One or more of the locations was not found.')
+            )
+
+        return [location.location_id for location in locations]
+
+    def clean(self):
+        cleaned_data = super().clean()
+
+        primary_location_id = cleaned_data['primary_location']
+        location_ids = cleaned_data['locations']
+        if primary_location_id:
```

Comment:
Nit and not needing addressing if you don't want to: Wouldn't `location_ids` always be valid id's at this point, since the code will raise `SQLLocation.DoesNotExist` if any invalid id (like `None`) is present? If this is the case, then we should be able to just do `if primary_location_id not in location_ids:`. If `primary_location_id` is `None`, it will never be in `location_ids`.

Reply:
This code mimics the way [mobile worker locations work](https://github.com/dimagi/commcare-hq/blob/0b793ca104a3fd80e4e2ee4d5902f04e05b09dff/corehq/apps/users/forms.py#L1349). You are right, if the user is using the form, with `LocationSelectWidget` and `PrimaryLocationWidget`, then the IDs will be valid. But I tend to agree with whoever wrote `CommtrackUserForm` for this to be a server-side validation layer.